### DISCUSSION TITLE
Face bluring feature

### DISCUSSION THREAD

**Discussion by Quasikot** [vladmandic/human#372] [General]
Hello! i want to fork your project and do the face bluring feature and maybe use it in my synth project for sentiment analisys. But i stuck on the building stage.
I myself have almost zero experience in java script, i work primarely with c++and python.
Please help me solve this issue.
I have working ssh connection with github because i use it.

When i do 
npm install --production=false

i get this output

npm WARN using --force I sure hope you know what you are doing.
npm WARN addRemoteGit Error: Command failed: git -c core.longpaths=true config --get remote.origin.url
npm WARN addRemoteGit
npm WARN addRemoteGit     at ChildProcess.exithandler (child_process.js:199:12)
npm WARN addRemoteGit     at emitTwo (events.js:106:13)
npm WARN addRemoteGit     at ChildProcess.emit (events.js:191:7)
npm WARN addRemoteGit     at maybeClose (internal/child_process.js:920:16)
npm WARN addRemoteGit     at Socket.<anonymous> (internal/child_process.js:351:11)
npm WARN addRemoteGit     at emitOne (events.js:96:13)
npm WARN addRemoteGit     at Socket.emit (events.js:188:7)
npm WARN addRemoteGit     at Pipe._handle.close [as _onclose] (net.js:509:12)
npm WARN addRemoteGit  vladmandic/tfjs resetting remote C:\Users\admin\AppData\Roaming\npm-cache\_git-remotes\git-github-com-vladmandic-tfjs-git-2874311b because of error: { Error: Command failed: git -c core.longpaths=true config --get remote.origin.url
npm WARN addRemoteGit
npm WARN addRemoteGit     at ChildProcess.exithandler (child_process.js:199:12)
npm WARN addRemoteGit     at emitTwo (events.js:106:13)
npm WARN addRemoteGit     at ChildProcess.emit (events.js:191:7)
npm WARN addRemoteGit     at maybeClose (internal/child_process.js:920:16)
npm WARN addRemoteGit     at Socket.<anonymous> (internal/child_process.js:351:11)
npm WARN addRemoteGit     at emitOne (events.js:96:13)
npm WARN addRemoteGit     at Socket.emit (events.js:188:7)
npm WARN addRemoteGit     at Pipe._handle.close [as _onclose] (net.js:509:12)
npm WARN addRemoteGit   killed: false,
npm WARN addRemoteGit   code: 1,
npm WARN addRemoteGit   signal: null,
npm WARN addRemoteGit   cmd: 'git -c core.longpaths=true config --get remote.origin.url' }
npm ERR! git clone --template=C:\Users\admin\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror git://github.com/vladmandic/tfjs.git C:\Users\admin\AppData\Roaming\npm-cache\_git-remotes\git-github-com-vladmandic-tfjs-git-2874311b: Cloning into bare repository 'C:\Users\admin\AppData\Roaming\npm-cache\_git-remotes\git-github-com-vladmandic-tfjs-git-2874311b'...
npm ERR! git clone --template=C:\Users\admin\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror git://github.com/vladmandic/tfjs.git C:\Users\admin\AppData\Roaming\npm-cache\_git-remotes\git-github-com-vladmandic-tfjs-git-2874311b: fatal: unable to connect to github.com:
npm ERR! git clone --template=C:\Users\admin\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror git://github.com/vladmandic/tfjs.git C:\Users\admin\AppData\Roaming\npm-cache\_git-remotes\git-github-com-vladmandic-tfjs-git-2874311b: github.com[0: 140.82.121.3]: errno=Unknown error
npm ERR! git rev-list -n1 master: fatal: ambiguous argument 'master': unknown revision or path not in the working tree.
npm ERR! git rev-list -n1 master: Use '--' to separate paths from revisions, like this:
npm ERR! git rev-list -n1 master: 'git <command> [<revision>...] -- [<file>...]'
npm ERR! git rev-list -n1 master:
npm ERR! git rev-list -n1 master: fatal: ambiguous argument 'master': unknown revision or path not in the working tree.
npm ERR! git rev-list -n1 master: Use '--' to separate paths from revisions, like this:
npm ERR! git rev-list -n1 master: 'git <command> [<revision>...] -- [<file>...]'
npm ERR! git rev-list -n1 master:
npm ERR! Windows_NT 10.0.22621
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\admin\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "install" "--production=false"
npm ERR! node v6.17.1
npm ERR! npm  v3.10.10
npm ERR! code 128

npm ERR! Command failed: git -c core.longpaths=true rev-list -n1 master
npm ERR! fatal: ambiguous argument 'master': unknown revision or path not in the working tree.
npm ERR! Use '--' to separate paths from revisions, like this:
npm ERR! 'git <command> [<revision>...] -- [<file>...]'
npm ERR!
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Projects\biometry\human\npm-debug.log

0 info it worked if it ends with ok
1 verbose cli [ 'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\Users\\admin\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'run',
1 verbose cli   'build' ]
2 info using npm@3.10.10
3 info using node@v6.17.1
4 verbose stack Error: ENOENT: no such file or directory, open 'C:\Projects\biometry\package.json'
4 verbose stack     at Error (native)
5 verbose cwd C:\Projects\biometry
6 error Windows_NT 10.0.22621
7 error argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\admin\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "run" "build"
8 error node v6.17.1
9 error npm  v3.10.10
10 error path C:\Projects\biometry\package.json
11 error code ENOENT
12 error errno -4058
13 error syscall open
14 error enoent ENOENT: no such file or directory, open 'C:\Projects\biometry\package.json'
15 error enoent ENOENT: no such file or directory, open 'C:\Projects\biometry\package.json'
15 error enoent This is most likely not a problem with npm itself
15 error enoent and is related to npm not being able to find a file.
16 verbose exit [ -4058, true ]

- Human library version?
[3.0.7]

- Browser or NodeJS and version (e.g. *NodeJS 14.15* or *Chrome 89*)?
try to build with v16.20.0-x64, v10.24.1-x64 node-v18.16.0-x64
- 
- OS and Hardware platform (e.g. *Windows 10*, *Ubuntu Linux on x64*, *Android 10*)?
 Windows 11

- Packager (if any) (e.g, *webpack*, *rollup*, *parcel*, *esbuild*, etc.)?
Dont know.

- Framework (if any) (e.g. *React*, *NextJS*, etc.)?
No

Package json

{
  "name": "@vladmandic/human",
  "version": "3.0.7",
  "description": "Human: AI-powered 3D Face Detection & Rotation Tracking, Face Description & Recognition, Body Pose Tracking, 3D Hand & Finger Tracking, Iris Analysis, Age & Gender & Emotion Prediction, Gesture Recognition",
  "sideEffects": false,
  "main": "dist/human.node.js",
  "module": "dist/human.esm.js",
  "browser": "dist/human.esm.js",
  "types": "types/human.d.ts",
  "exports": {
    "node": {
      "require": "./dist/human.node.js",
      "import": "./dist/human.node.js",
      "module": "./dist/human.node.js"
    },
    "require": "./dist/human.node.js",
    "import": "./dist/human.esm.js",
    "script": "./dist/human.js",
    "module": "./dist/human.esm.js",
    "types": "./types/human.d.ts",
    "dist/human": "./dist/human.js",
    "dist/human.js": "./dist/human.js",
    "dist/human.esm": "./dist/human.esm.js",
    "dist/human.esm.js": "./dist/human.esm.js",
    "dist/human.esm-nobundle": "./dist/human.esm-nobundle.js",
    "dist/human.esm-nobundle.js": "./dist/human.esm-nobundle.js",
    "dist/human.node": "./dist/human.node.js",
    "dist/human.node.js": "./dist/human.node.js",
    "dist/human.node-wasm": "./dist/human.node-wasm.js",
    "dist/human.node-wasm.js": "./dist/human.node-wasm.js",
    "dist/human.node-gpu": "./dist/human.node-gpu.js",
    "dist/human.node-gpu.js": "./dist/human.node-gpu.js"
  },
  "author": "Vladimir Mandic <mandic00@live.com>",
  "bugs": {
    "url": "https://github.com/vladmandic/human/issues"
  },
  "homepage": "https://vladmandic.github.io/human/demo/index.html",
  "license": "MIT",
  "engines": {
    "node": ">=14.0.0"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/vladmandic/human.git"
  },
  "scripts": {
    "start": "node --no-warnings demo/nodejs/node.js",
    "dev": "build --profile development",
    "clean": "build --profile clean",
    "build": "rimraf test/build.log && node build.js",
    "test": "node --no-warnings --unhandled-rejections=strict --trace-uncaught test/node.js",
    "lint": "eslint *.json *.js src demo test models wiki",
    "scan": "npx auditjs@latest ossi --dev --quiet"
  },
  "keywords": [
    "human",
    "human-library",
    "face-detection",
    "faceid",
    "face-geometry",
    "face-embedding",
    "face-recognition",
    "face-description",
    "face-matching",
    "body-tracking",
    "body-segmentation",
    "hand-tracking",
    "iris-tracking",
    "age-estimation",
    "emotion-detection",
    "gender-prediction",
    "gesture-recognition",
    "gaze-tracking",
    "age-gender",
    "tensorflowjs",
    "tfjs",
    "tensorflow"
  ],
  "devDependencies": {
    "@html-eslint/eslint-plugin": "^0.19.0",
    "@html-eslint/parser": "^0.19.0",
    "@microsoft/api-extractor": "^7.35.2",
    "@tensorflow/tfjs-backend-cpu": "^4.7.0",
    "@tensorflow/tfjs-backend-wasm": "^4.7.0",
    "@tensorflow/tfjs-backend-webgl": "^4.7.0",
    "@tensorflow/tfjs-backend-webgpu": "4.7.0",
    "@tensorflow/tfjs-converter": "^4.7.0",
    "@tensorflow/tfjs-core": "^4.7.0",
    "@tensorflow/tfjs-data": "^4.7.0",
    "@tensorflow/tfjs-layers": "^4.7.0",
    "@tensorflow/tfjs-node": "^4.7.0",
    "@tensorflow/tfjs-node-gpu": "^4.7.0",
    "@types/node": "^20.3.0",
    "@types/offscreencanvas": "^2019.7.0",
    "@typescript-eslint/eslint-plugin": "^5.59.9",
    "@typescript-eslint/parser": "^5.59.9",
    "@vladmandic/build": "^0.9.2",
    "@vladmandic/pilogger": "^0.4.8",
    "@vladmandic/tfjs": "github:vladmandic/tfjs",
    "canvas": "^2.11.2",
    "esbuild": "^0.18.1",
    "eslint": "8.42.0",
    "eslint-config-airbnb-base": "^15.0.0",
    "eslint-plugin-html": "^7.1.0",
    "eslint-plugin-import": "^2.27.5",
    "eslint-plugin-json": "^3.1.0",
    "eslint-plugin-markdown": "^3.0.0",
    "eslint-plugin-node": "^11.1.0",
    "eslint-plugin-promise": "^6.1.1",
    "rimraf": "^5.0.1",
    "tslib": "^2.5.3",
    "typedoc": "0.24.8",
    "typescript": "5.1.3"
  }
}

**Comment by vladmandic** [comment]
> 14 error enoent ENOENT: no such file or directory, open 'C:\Projects\biometry\package.json'

you don't have project created at all.  
and you've posted `package.json` from `human` library instead.  

you need to step back and check js/npm ecosystem basics before diving into `human`

**Reply by Quasikot** [reply]
ok step back one one hierarhy level. I get it. Thank you!
